tests/test_main.py

```diff
@@ -6,26 +6,28 @@
 from unittest.mock import patch
 
 from devenv import main
+from devenv.lib import config
+from devenv.lib.config import DEFAULT_CONFIG
 from tests.utils import chdir
 
 
-def test(tmp_path: str) -> None:
-    with patch("devenv.main.CI", True), patch(
+def _test(tmp_path: str) -> None:
+    with patch("devenv.constants.CI", True), patch(
         "devenv.bootstrap.main"
     ) as mock_bootstrap, patch("devenv.sync.main") as mock_sync:
         config_path = f"{tmp_path}/.config/sentry-devenv/config.ini"
         coderoot = f"{tmp_path}/code"
-        main.DEFAULT_CONFIG["devenv"]["coderoot"] = coderoot
+        DEFAULT_CONFIG["devenv"]["coderoot"] = coderoot
```

Comment:
this irreversibly changes the global variable -- you probably want:

```python
with mock.patch.dict(DEFAULT_CONFIG["devenv"], {"coderoot": coderoot}):
   ...
```

this will clean itself up after the block is exited

Reply:
Ah.. True.  I've removed all of these shenanigans in a later PR.

Reply:
New test doesn't contain any mutation on global objects.

`def test(tmp_path: str) -> None:
    with patch("devenv.constants.CI", True):
        config_path = f"{tmp_path}/.config/sentry-devenv/config.ini"
        coderoot = f"{tmp_path}/code"

        assert not os.path.exists(config_path)
        main.devenv(
            ("/path/to/argv0", "bootstrap", "-d", f"coderoot:{coderoot}"),
            config_path,
        )
        assert os.path.exists(config_path)

        config = read_config(config_path)
        assert config.get("devenv", "coderoot") == coderoot`

Reply:
maybe we should delete the test here and reintroduce it later -- that'd be more clear I think